src/app/member/components/MemberEditPage.tsx

```diff
@@ -0,0 +1,361 @@
+    .string({ message: "Name is required" })
+    .min(10, "Name should have at least 10 characters"),
+  role: z.nativeEnum(UserRole, {
+    errorMap: () => {
+      return { message: "Please select your user type" };
+    },
+  }),
+  ehad_start_date: z
+    .date()
+    .min(addDays(new Date(), 30), "Minmum Ehad Duration is one Month"),
+});
+
+export const UserForm = () => {
+  const form = useForm<z.infer<typeof formSchema>>({
+    resolver: zodResolver(formSchema),
+    defaultValues: {
+      name: "",
+      phone: "",
+      date_of_birth: new Date(),
+      cnic: "",
+      shift: undefined,
+      address: "",
+      role: undefined,
+      ehad_start_date: new Date(),
+    },
+    mode: "onBlur",
+  });
+  const handleSubmittion = (values: z.infer<typeof formSchema>) => {
+    onSubmit(values);
+    form.reset();
+  };
+  const {
+    formState: { errors },
+  } = form;
+  return (
+    <FormWrapper>
+      <Form {...form}>
+        <form
+          onSubmit={form.handleSubmit(handleSubmittion)}
+          className="space-y-4  "
+        >
+          <h4 className="text-xl text-black text-bold">Information</h4>
+          <FormField
+            control={form.control}
+            name="name"
+            render={({ field }) => (
+              <FormItem className="flex flex-col">
+                <Label>Name</Label>
+
+                <Input
+                  placeholder="Enter Your Name"
+                  {...field}
+                  data-testid="name"
+                  className={
+                    errors?.name &&
+                    "border-red-500 border focus-visible:ring-0 "
+                  }
+                />
+
+                <FormMessage />
+              </FormItem>
+            )}
+          />
+          <FormField
+            control={form.control}
+            name="phone"
+            render={({ field }) => (
+              <FormItem className="flex flex-col">
+                <Label>Phone number</Label>
+                <FormControl>
+                  <Input
+                    placeholder="0300-0000000"
+                    {...field}
+                    data-testid="phone"
+                    className={
+                      errors?.phone &&
+                      "border-red-500 border focus-visible:ring-0 "
+                    }
+                  />
+                </FormControl>
+                <FormMessage />
+              </FormItem>
+            )}
+          />
+          <FormField
+            control={form.control}
+            name="cnic"
+            render={({ field }) => (
+              <FormItem className="flex flex-col">
+                <Label>CNIC</Label>
+                <FormControl>
+                  <Input
+                    placeholder="30000-0000000-0"
+                    {...field}
+                    data-testid="cnic"
+                    className={
+                      errors?.cnic &&
+                      "border-red-500 border focus-visible:ring-0 "
```

Comment:
I think there is need no add custom styling for error because useForm hook handle it according to zod validation.

Reply:
Actually when error occured the border of the input was not showing red border so i added it custom styling